code/parse/sexp.cpp

```diff
@@ -16458,13 +16487,30 @@ void sexp_set_asteroid_field(int n)
 	}
 
 	SCP_vector<SCP_string> targets;
-	if (n >= 0) {
-		for (; n >= 0; true ? n = CDR(n) : n = -1) {
-			auto ship_entry = eval_ship(n);
-			if (!ship_entry)
-				continue;
+	if (!new_sexp) {
+		if (n >= 0) {
+			for (; n >= 0; true ? n = CDR(n) : n = -1) {
```

Comment:
``true ? n = CDR(n) : n = -1`` seems weird.
That just boils down to ``n = CDR(n)``. Either it should probably be simplified to that, or something may be amiss with the conditional here.

Reply:
I'm not sure what I was smoking when I put that there. I'm sure I copied it from the old asteroid sexp but given I wrote that too, I'm still not sure. Bet I copied something similar from somewhere else, modified it for testing, and promptly forgot about it.

Either way, yes you're right. Fixed.